testing/integration-testing-hooks.ts

```diff
@@ -458,7 +461,17 @@ export function skipIfEnvServerVersion(semverCondition: string): void {
   before(async function() {
     let testServerVersion = process.env.MONGOSH_SERVER_TEST_VERSION;
     if (!testServerVersion) {
-      testServerVersion = await getInstalledMongodVersion();
+      try {
+        testServerVersion = await getInstalledMongodVersion();
+      } catch(e) {
+        // no explicitly specified version but also no local mongod installation
+        // -> spawn a server to get the "live" version
+        console.log('Starting server to detect runtime version...');
+        const server = startTestServer('shared');
+        await server.start();
+        skipIfServerVersion(server, semverCondition);
```

Comment:
this runs a `before` inside a `before`, how is it going to work?

Reply:
Good question 😅 It did work locally tho - I'd assume since it's still in "before" the other before hook gets registered and executed. With your suggestion below it would be a lot simpler and not required to launch it here.

Reply:
Obscure mocha voodoo, well if it works .. :)